src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -167,7 +176,10 @@ public static function createConnection($dsn, array $options = array())
             $params['scheme'] = $scheme;
             $params['database'] = $params['dbindex'] ?: null;
             $params['password'] = $auth;
-            $redis = new $class((new Factory())->create($params));
+
+            $predisOptions = false !== isset($params['predis_options']) && true === \is_array($params['predis_options']) ? $params['predis_options'] : array();
```

Comment:
no need for the boolean comparisons:
`$predisOptions = isset($params['predis_options']) && \is_array($params['predis_options']) ? $params['predis_options'] : array();`
but shouldn't we throw when predis_options is not an array?